code/modules/mob/living/carbon/human/human_damage.dm

```diff
@@ -151,6 +145,41 @@
 				O.unmutate()
 				src << "<span class = 'notice'>Your [O.name] is shaped normally again.</span>"
 	hud_updateflag |= 1 << HEALTH_HUD
+	
+// Defined here solely to take species flags into account without having to recast at mob/living level.
+/mob/living/carbon/human/getOxyLoss()
+	if(species.flags & NO_BREATHE)
+		oxyloss = 0
+	return ..()
+
+/mob/living/carbon/human/adjustOxyLoss(var/amount)
+	if(species.flags & NO_BREATHE)
+		oxyloss = 0
+	else
+		..()
+
+/mob/living/carbon/human/setOxyLoss(var/amount)
+	if(species.flags & NO_BREATHE)
+		oxyloss = 0
+	else
+		..()
+
+/mob/living/carbon/human/getToxLoss()
+	if(species.flags & NO_POISON)
+		toxloss = 0
+	return ..()
+
+/mob/living/carbon/human/adjustToxLoss(var/amount)
+	if(species.flags & NO_POISON)
+		toxloss = 0
+	else
+		..()
+
+/mob/living/carbon/human/setToxLoss(var/amount)
+	if(species.flags & NO_POISON)
+		toxloss = 0
+	else
+		..()

```

Comment:
Minor suggestion, you may just want to bring all damage types in line for species to use either damage multipliers or bit flags, instead of having 2 damage types use bit flags for immunity, and two using damage mods.